crates/volta-core/src/toolchain/mod.rs

```diff
@@ -40,7 +38,7 @@ impl LazyToolchain {
 }
 
 pub struct Toolchain {
-    platform: Option<PlatformSpec>,
+    platform: Option<Rc<DefaultPlatformSpec>>,
```

Comment:
In this approach, why is the `Toolchain.platform` *always* `DefaultPlatformSpec`? Is there no scenario where the toolchain platform spec is the *project* value? (I'm trying to reload the `Toolchain` context into my head so this may be a stupid question!)

Reply:
It's not a stupid question, it's actually a consequence of poor naming :smile: The `Toolchain` object really represents the "Default Toolchain", it serializes and deserializes the `platform.json` that we write into the user's `~/.volta` directory. This is in contrast with the `Project`, which represents the information we pull from `package.json`.

So, for example, when a user does `volta install node`, we call `Toolchain::set_active_node`, while when a user does `volta pin node`, we call `Project::pin_node`